src/utils.ts

```diff
@@ -68,12 +68,18 @@ export function getQuery (input: string): QueryObject {
   return parseQuery(parseURL(input).search)
 }
 
+export function isRelativeUrl (input: string): boolean {
+  return ['./', '../'].some(part => input.startsWith(part))
+}
+
 export function joinURL (base: string, ...input: string[]): string {
   let url = base || ''
 
   for (const i of input) {
     const part = withoutLeadingSlash(i)
-    if (part !== '/') {
+    if (!url && isRelativeUrl(i)) {
```

Comment:
Tests will pass with `!url` test only...

Handling relative urls seems a good idea but i guess would be better implemented in another (feature)-pr and considered for all utils other than join